document/introduction/index.rst

```diff
@@ -1,6 +1,54 @@
 Introduction
 ============
 
-.. todo::
+WebAssembly (abbreviated Wasm [#wasm]_) is a *safe, portable, low-level code format*
+designed for efficient execution and compact representation.
+Its main goal is to enable high performance applications on the Web, but it does not make any Web-specific assumptions or provide Web-specific features, so can be employed in other environments as well.
 
-   All there is to say
+WebAssembly is an open standard developed by a `W3C Community Group <https://www.w3.org/community/webassembly/>`_ that includes representatives of all major browser vendors.
+
+This document describes version |release| of the standard.
+It is intended that it will be superseded by new incremental releases with additional features in the future.
+
+
+Design Goals
+------------
+
+.. index:: design goals, portability
+
+The design goals of WebAssembly are the following:
+
+* Fast, safe, and portable *semantics*:
+
+  * **Fast**: executes with near native code performance, taking advantage of capabilities common to all contemporary hardware.
+
+  * **Safe**: code is validated and executes in a memory-safe, sandboxed environment preventing data corruption or security breaches.
```

Comment:
Wasm programs cannot access memory in unspecified ways, and even the attempt of doing so (for linear memories) does not lead to undefined behaviour or crashes. Memory safety is the established term for that, so I'm not sure what the confusion would be. It is what necessitates validation, bounds checks, checked calls, etc. Not mentioning it here would completely miss the motivation for all of these. It's different from sandboxing, which is about controlling access to host resources and is much more coarse-grained.